types/query/filtered_pagination.go

```diff
@@ -102,7 +102,9 @@ func FilteredPaginate(
 		}
 
 		if numHits == end+1 {
-			nextKey = iterator.Key()
+			if nextKey == nil {
```

Comment:
if there are multiple queries each have 100 to list all 1000 entries, shouldn't this block further query iteration since user doesn't know which is the next key?